templates/server/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,30 @@
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.LinkOption
+
+// the path where the project got generated
+Path projectPath = Paths.get(request.outputDirectory, request.artifactId)
+
+// the properties available to the archetype
+Properties properties = request.properties
+String batch = properties.get("batch")
+
+if (batch != "batch") {
+  // delete folder batch recursively
+  Path rootPath = projectPath.resolve("batch")
+  deleteDirectoryRecursion(rootPath)
+}
+
+void deleteDirectoryRecursion(Path path) {
+  if (Files.isDirectory(path, LinkOption.NOFOLLOW_LINKS)) {
+    try {
+      for (Path entry : Files.newDirectoryStream(path)) {
+        deleteDirectoryRecursion(entry);
```

Comment:
as you make use of omitting semicolons as it seems best practice in groovy, it would be even nicer, if you do that consistently (including line 29).